GitUI/BranchTreePanel/RepoObjectsTree.SettingsContextMenu.cs

```diff
@@ -5,18 +5,41 @@ namespace GitUI.BranchTreePanel
 {
     public partial class RepoObjectsTree
     {
+        private void ShowEnabledTrees()
+        {
+            if (tsmiShowBranches.Checked)
+            {
+                AddTree(_branchesTree, 0);
+            }
+
+            if (tsmiShowRemotes.Checked)
+            {
+                AddTree(_remotesTree, 1);
+            }
+
+            if (tsmiShowTags.Checked)
+            {
+                AddTree(_tagTree, 2);
+            }
+
+            if (tsmiShowSubmodules.Checked)
+            {
+                AddTree(_submoduleTree, 3);
+            }
+        }
+
         private void tsmiShowBranches_Click(object sender, EventArgs e)
         {
             AppSettings.RepoObjectsTreeShowBranches = tsmiShowBranches.Checked;
             _searchResult = null;
             if (tsmiShowBranches.Checked)
             {
-                AddBranches();
+                AddTree(_branchesTree, 0);
```

Comment:
I would add a private enum to track the order, it looks like we use magic numbers in few places.
It could lead to bugs should we wish to modify the order.

Reply:
You may be right, but actually for now I would leave it as is. There is exactly one place, now, where we specify the index, and that's when we call AddTree. When working on the feature of reordering of root nodes, we can decide the best way to implement this. Truth is, if we use an enum now, like "Branches = 0, Remotes = 1, Tags = 2, Submodules = 3", then it would be difficult to reorder.